src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -71,58 +68,149 @@ class F extends ValidationTest {
+    { size: [31, 32], mipLevelCount: 7, _success: false }, // too big mip chains on width for 2D texture are disallowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1, ?x?)
+    { size: [32, 31], mipLevelCount: 7, _success: false }, // too big mip chains on height for 2D texture are disallowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1, ?x?)
+    { size: [31, 31], mipLevelCount: 5, _success: true }, // full mip chains for non-power-of-two 2D texture are allowed (Mip level sizes: 31x31, 15x15, 7x7, 3x3, 1x1)
+    { size: [31, 31], mipLevelCount: 6, _success: false }, // too big mip chains for non-power-of-two 2D texture are disallowed (Mip level sizes: 31x31, 15x15, 7x7, 3x3, 1x1, ?x?)
+    { size: [32, 1, 1], dimension: '1d' as const, mipLevelCount: 6, _success: true }, // full mip chains for 1D texture are allowed (Mip level sizes: 32, 16, 8, 4, 2, 1)
+    { size: [32, 1, 1], dimension: '1d' as const, mipLevelCount: 7, _success: false }, // too big mip chains for 1D texture are not allowed (Mip level sizes: 32, 16, 8, 4, 2, 1)
+    { size: [31, 1, 1], dimension: '1d' as const, mipLevelCount: 5, _success: true }, // full mip chains for non-power-of-two 1D texture are allowed (Mip level sizes: 31, 15, 7, 3, 1)
+    { size: [31, 1, 1], dimension: '1d' as const, mipLevelCount: 6, _success: false }, // too big mip chains for non-power-of-two 1D texture are not allowed (Mip level sizes: 31, 15, 7, 3, 1)
+    { size: [32, 32, 32], dimension: '3d' as const, mipLevelCount: 6, _success: true }, // full mip chains for 3D texture are allowed (Mip level sizes: 32x32x32, 16x16x16, 8x8x8, 4x4x4, 2x2x2, 1x1x1)
+    { size: [32, 32, 32], dimension: '3d' as const, mipLevelCount: 7, _success: false }, // too big mip chains for 3D texture are not allowed (Mip level sizes: 32x32x32, 16x16x16, 8x8x8, 4x4x4, 2x2x2, 1x1x1)
+    { size: [31, 31, 31], dimension: '3d' as const, mipLevelCount: 5, _success: true }, // full mip chains for non-power-of-two 3D texture are allowed (Mip level sizes: 31x31x31, 15x15x15, 7x7x7, 3x3x3, 1x1x1)
+    { size: [31, 31, 31], dimension: '3d' as const, mipLevelCount: 6, _success: false }, // too big mip chains for non-power-of-two 3D texture are not allowed (Mip level sizes: 31x31x31, 15x15x15, 7x7x7, 3x3x3, 1x1x1)
+    { size: [32, 32], mipLevelCount: 100, _success: false }, // undefined shift check if miplevel is bigger than the integer bit width
+    { size: [32, 8], mipLevelCount: 6, _success: true }, // non square mip map halves the resolution until a 1x1 dimension. (Mip maps: 32x8, 16x4, 8x2, 4x1, 2x1, 1x1)
+    { size: [32, 8], mipLevelCount: 7, _success: true }, // too big mip chains for non square mip map halves are disallowed (Mip maps: 32x8, 16x4, 8x2, 4x1, 2x1, 1x1)
+    { size: [32, 32, 64], mipLevelCount: 7, _success: false }, // array layer count for 2D texture should not be taken account to calculate mip levels. (Mip maps: 32x32x64, 16x16x64, 8x8x64, 4x4x64, 2x2x64, 1x1x64)
```

Comment:
nit: Should go to `?x?x64`

And please do the same fix for other comments with the same problem.

Reply:
Done.